### ISSUE TITLE
Support for cluster auto scaling

### ISSUE THREAD

**Issue by laurentpellegrino** [hcloud-talos/terraform-hcloud-talos#16]
It would be greater if autoscaling is supported with autoscaling agents such as in terraform-hcloud-kube-hetzner

**Comment by mrclrchtr** [comment]
Yes, that would be a great feature. Unfortunately, it is not currently planned. Maybe I will implement it in the future. PRs are welcome.

**Comment by hegerdes** [comment]
Autoscaling with terraform does not work unless the infrastructure provider offers a dedicated resource for it like aws node-pool or azure vmss. Hetzner does not provide such a thing unfortunately so terrafrom tracks the VMs all individually.

But you can still setup auto-scale within Kubernetes and its super easy for Talos. Just put the talos worker machine config with some extra args into a save place (like external secret store or k8s secret) and deploy k8s-cluster-autoscaler and tell it to use the conf. The cluster autoscaler has native support for Hetzner.

I create the conf like this:
https://github.com/hegerdes/GitOps/blob/main/infra/hcloud-tf-k8s-talos/main.tf#L57-L75

After that I deploy the autoscaler and make it use the config via the `HCLOUD_CLUSTER_CONFIG` env.

**Comment by mrclrchtr** [comment]
Many thanks for the insights. This will help others to realize autoscaling. 

Here is more context https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/hetzner/README.md

I think we can close the issue for now.

**Comment by dihmeetree** [comment]
Hey @hegerdes thanks for the link to your implementation of the Cluster Autoscaler for Hetzner. Any way you'd be willing to get a PR going with the autoscaler addition to this project perhaps? Just discovered this project and it's awesome! The only thing I think that's missing is just cluster autoscaling feature (for autoscaling workers) 🙂
